resource/Resource.md

```diff
@@ -0,0 +1,133 @@
+# Resource API Overview
+The resource library primarily defines a type that captures about information about the entity
+for which stats or traces are recorded. It further provides a framework for detection of
+resource information from the environment and progressive population as signals propagate
+from the core instrumentation library to the a backend's exporter.
+
+## Resource type
+A `Resource` describes the entity for which a signal was collected through two fields:
+* `type`: a string which describes a well-known type of entity. It SHOULD be namespaced
+to avoid collisions across different environment, e.g. `kubernetes.io/container`,
+`cloud.google.com/gce/instance`. 
+* `labels`: a dictionary of labels with string keys and values that provide information
+about the entity.
+
+Type strings and label keys MAY start with a domain name and MAY be further namespaced through
+single slashes. All other characters MUST be alphanumeric. Label values MAY contain any valid
+unicode character.
+
+Implementations MAY define a `Resource` data type, constructed from the parameters above.
+Resource MAY have getters for retrieving all the information used in `Resource` definition.
+
+Example in Go:
+```go
+type Resource {
+	Type   string
+	Labels map[string]string
+}
+```
+
+## Populating resources
+Resource information MAY be populated at any point between startup of the instrumented
+application and passing it the a backend-specific exporter. This explicitly includes
+the path through future OpenCensus components such as agents or services.
+
+For example, process-identifying information may be populated through the library while
+an agent attaches further labels about the underlying VM, the cluster, or geo-location.
+
+### From environment variables
+Population of resource information from environment varibales MUST be provided by the
+core library. It provides the user with an ubiquitious way to manually provide information
+that may not be detectable automatically through available integration libraries.
+
+Two environment varibales are used:
+* `OC_RESOURCE_TYPE`: defines the resource type. Leading and trailing whitespaces are trimmed.
+* `OC_RESOURCE_LABELS`: defines resource labels as a comma-seperated list of key/value pairs.
+The key MUST be seperated from the value by a single `=`. Values MAY be quoted with a single
+leading and trailing `"`. If a values contains whitespaces, `=`, or `"` characters it MUST be
+quoted and `"` characters MUST be escaped.
+
```

Comment:
Can you please give examples of OC_RESOURCE_TYPE and OC_RESOURCE_LABELS to demonstrate the format?

Reply:
Done.